### PR TITLE
upgrade anchor

### PR DESCRIPTION
**Pull request by dnut**
- get anchor to latest 0.25
- which also pushes solana to 1.10
- couple other tweaks needed to make cargo update happy (tokio and bytes)
- some rust code fixes were also needed for breaking changes in the libraries

### CODE REVIEW

File: .github/workflows/devnet.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ on:
   workflow_dispatch: {}
 
 env:
-  ANCHOR_CLI_VERSION: 0.24.2
+  ANCHOR_CLI_VERSION: 0.25.0
```
Comment (suscd) [inline .github/workflows/devnet.yaml:13]: the latest published version of anchor is `0.24.2`, so this doesn't work and breaks the integration test workflow which is falsely passing even though no tests are being run.

**Review by suscd — CHANGES_REQUESTED** [review]

File: .github/workflows/devnet.yaml (hunk 1 above)
Comment (dnut) [reply .github/workflows/devnet.yaml:13]: oops i assumed 0.25 was fully released since it was released on github

Comment (dnut) [reply .github/workflows/devnet.yaml:13]: i see 0.25.0 here https://crates.io/crates/anchor-cli

Comment (suscd) [reply .github/workflows/devnet.yaml:13]: I also see there's a `0.25` version for the npm package: https://www.npmjs.com/package/@project-serum/anchor

seems like the problem is just running the latest anchor cli

Comment (suscd) [reply .github/workflows/devnet.yaml]: It's probably fine to leave the cli version at 0.24.2 for now, while the actual linked library is 0.25.

**Review by suscd — APPROVED** [review]

**Pull request merged**